aptos-move/framework/src/built_package.rs

```diff
@@ -172,20 +179,11 @@ impl BuiltPackage {
         };
 
         eprintln!("Compiling, may take a little while to download git dependencies...");
-        let mut package = build_config.compile_package_no_exit(&package_path, &mut stderr())?;
+        let (mut package, model_opt) =
+            build_config.compile_package_no_exit(&package_path, &mut stderr())?;
 
-        // Build the Move model for extra processing and run extended checks as well derive
-        // runtime metadata
-        let model = &build_model(
-            options.dev,
-            package_path.as_path(),
-            options.named_addresses.clone(),
-            None,
-            bytecode_version,
-            compiler_version,
-            skip_attribute_checks,
-            options.known_attributes.clone(),
-        )?;
+        // Run extended checks as well derive runtime metadata
+        let model = &model_opt.expect("move model");
         let runtime_metadata = extended_checks::run_extended_checks(model);
```

Comment:
Is it possible to do these steps during compilation instead of afterward, for example via visitors?

Reply:
Not in the compiler v1 architecture.